librust/proc_macro/rust/bridge/punct.rs

```diff
@@ -1,16 +1,23 @@
 use bridge::span::Span;
+use std::convert::TryInto;
+use std::ffi::c_uchar;
 use Spacing;
 
 #[repr(C)]
 #[derive(Clone, Debug)]
 pub struct Punct {
-    pub(crate) ch: char,
+    pub(crate) ch: c_uchar,
```

Comment:
A rust char is an u32 with a range of all unicode scalar values. A C char is a u8 which is unable to represent arbitrary unicode scalar values.

Reply:
> By the way do you want me to keep reviewing as you are implementing it or do you prefer if I wait until this is no longer a draft?

To be fair, the way to achieve this still contains some dark spots. I really appreciate those early comments as they help me  to either confirm I'm going in the right direction or that the code/design is flawed.